src/Bicep.Cli.IntegrationTests/CliScenarioTests.cs

```diff
@@ -187,5 +189,58 @@ param roleAssignmentName string
 
                 """);
         }
+
+        [TestMethod]
+        public async Task Test_Issue13785()
+        {
+            var paramFile =
+              """
+              {
+                "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+                "contentVersion": "1.0.0.0",
+                "parameters": {
+                  "foo": {
+                    "value": "bar"
+                  }
+                }
+              }
+              """;
+            var expectedOutput =
+                """
+                using '../main.bicep'
+
+                param foo = 'bar'
+
+                """;
+
+            var (jsonPath, bicepparamPath) = Setup(TestContext, paramFile);
+            Directory.SetCurrentDirectory(Path.GetDirectoryName(jsonPath)!);
+            var (output, _, result) = await Bicep("decompile-params", jsonPath, "--bicep-file", "../main.bicep");
+
+            using (new AssertionScope())
+            {
+                output.Should().BeEmpty();
+                result.Should().Be(0);
+                File.ReadAllText(bicepparamPath).Should().BeEquivalentToIgnoringNewlines(expectedOutput);
+            }
+        }
+
+        private static (string jsonPath, string bicepparamPath) Setup(TestContext context, string template, string? inputFile = null, string? outputDir = null)
+        {
+            var jsonPath = FileHelper.SaveResultFile(context, inputFile ?? "param.json", template);
+
+            string bicepparamPath;
+
+            if (outputDir is null)
+            {
+                bicepparamPath = PathHelper.GetDefaultDecompileparamOutputPath(jsonPath);
+            }
+            else
+            {
+                bicepparamPath = FileHelper.GetResultFilePath(context, outputDir);
+            }
```

Comment:
[nit]: This is a personal preference, but I would use a ternary expression instead of a single assignment over multiple statements.